src/main.ts

```diff
@@ -85,21 +85,19 @@ function _getAndValidateArgs(): IIssuesProcessorOptions {
     'operations-per-run'
   ]) {
     if (isNaN(parseInt(core.getInput(numberInput)))) {
-      core.setFailed(
-        new Error(`Option "${numberInput}" did not parse to a valid integer`)
-      );
+      const errorMessage = `Option "${numberInput}" did not parse to a valid integer`;
+      core.setFailed(errorMessage);
+      throw new Error(errorMessage);
     }
   }
 
   for (const optionalDateInput of ['start-date']) {
     // Ignore empty dates because it is considered as the right type for a default value (so a valid one)
     if (core.getInput(optionalDateInput) !== '') {
       if (!isValidDate(new Date(core.getInput(optionalDateInput)))) {
-        core.setFailed(
-          new Error(
-            `Option "${optionalDateInput}" did not parse to a valid date`
-          )
-        );
+        const errorMessage = `Option "${optionalDateInput}" did not parse to a valid date`;
+        core.setFailed(errorMessage);
+        throw new Error(errorMessage);
```

Comment:
To give some context, not so long ago I provided a PR to use `core.setFailed` instead of `throw`.
Then I was doubtful about the fact that internally it throws an error but looking back at the doc again https://github.com/actions/toolkit/tree/main/packages/core#exit-codes mention an exit code so in that case I will remove this, nice catch

Reply:
783c103202fc7af585fcb6ff1b3d005ff59b89de